src/store/ShareStore.js

```diff
@@ -64,17 +75,17 @@ export default class ShareStore {
         hashPromise
             .then((hash) => {
                 if (hash) {
-                    self.shareLink = `https://bit.ly/${hash}`;
-                    self.urlGenerated = true;
+                    self.setShareLink(`https://bit.ly/${hash}`);
+                    self.setUrlGenerated(true);
                 } else {
-                    self.shortUrlFailed = true;
-                    self.urlGenerated = false;
+                    self.setShortUrlFailed(true);
+                    self.setUrlGenerated(false);
                 }
-                self.loading = false;
+                self.setLoading(false);
             })
             .catch(() => {
-                self.loading = false;
-                self.urlGenerated = false;
+                self.setLoading(false);
+                self.setUrlGenerated(false);
```

Comment:
the point of using actions was to batch the mutations and have the render execute only when all mutations within an action is complete. Doing this way is not that much different from mutating each observable separately. TLDR if there are multiple mutations in one batch, group them to one action.